### DISCUSSION TITLE
Custom fenced extension issues

### DISCUSSION THREAD

**Discussion by agentphoenix** [thephpleague/commonmark#637] [Q&A] [answered]
I've been working recently on building an extension to add alerts for a documentation site I'm building. I started with fenced code blocks as an example as that's the syntax I want to use:

```
:::note
This is my note with some **extra** content.
:::
```
The only way I was able to get this working was to set `isCode` to `true`. (When I didn't, I wasn't ever able to differentiate between the opening fence and closing fence, it just saw it as 2 opening fences and rendered multiple elements.)

Everything is actually working now, but I want the contents of the alert to be parsed as normal Markdown so I can add links and put some basic styling in, but nothing I've done gets that working. I've looked at other parsers/renderers and tried a bunch of things from those classes, but nothing has worked so far. Is there something I'm missing to get that working? Is someone able to point me in the right direction?

Here's the work I've done so far on the extension:

https://gist.github.com/agentphoenix/64d48899d56b892011832dcce0b28af8

**Comment by agentphoenix** [comment, marked as answer]
I ended up figuring out what I was doing wrong. I just had to push the remaining contents through the `renderInlines` and everything started working!

**Reply by linushstge** [reply]
Could you please share your final extension? 😊
